common/txmgr/inmemory_store.go

```diff
@@ -210,6 +212,24 @@ func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) FindT
 
 // UpdateBroadcastAts updates the broadcast_at time for a given set of attempts
 func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) UpdateBroadcastAts(ctx context.Context, now time.Time, txIDs []int64) error {
```

Comment:
If we're going to let callers pass in the time, then we shouldn't assume it will be `now`. Otherwise we should just call `time.Now()` inside this method.
```suggestion
func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) UpdateBroadcastAts(ctx context.Context, time time.Time, txIDs []int64) error {
```